case/src/main/kotlin/com/ritense/case_/widget/table/TableCaseWidgetDataProvider.kt

```diff
@@ -0,0 +1,86 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.case_.widget.table
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.module.kotlin.treeToValue
+import com.jayway.jsonpath.Configuration
+import com.jayway.jsonpath.JsonPath
+import com.jayway.jsonpath.Option
+import com.ritense.case_.domain.tab.CaseWidgetTab
+import com.ritense.case_.widget.CaseWidgetDataProvider
+import com.ritense.valueresolver.ValueResolverService
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.Pageable
+import java.util.UUID
+
+class TableCaseWidgetDataProvider(
+    private val objectMapper: ObjectMapper,
+    private val valueResolverService: ValueResolverService
+) : CaseWidgetDataProvider<TableCaseWidget> {
+
+    override fun supportedWidgetType() = TableCaseWidget::class.java
+
+    override fun getData(documentId: UUID, widgetTab: CaseWidgetTab, widget: TableCaseWidget, pageable: Pageable): Page<Map<String, Any?>> {
+        val collectionPlaceholder =
+            valueResolverService.resolveValues(documentId.toString(), listOf(widget.properties.collection))[widget.properties.collection]
+        val collection = objectMapper.valueToTree<JsonNode>(collectionPlaceholder)
+        if (collection !is ArrayNode) {
+            throw InvalidCollectionException()
+        }
+
+        val pagedCollection = collection.chunked(
+            pageable.pageSize
+        )
+
+        val result = pagedCollection.getOrElse(pageable.pageNumber, defaultValue = { _ -> listOf() })
+            .onEachIndexed { index, node ->
+                if (!node.isContainerNode) {
+                    throw InvalidCollectionNodeTypeException(index)
+                }
+            }.map { child ->
+                widget.properties.columns.associate { column ->
+                    val value = if (column.value.startsWith("$")) {
+                        JSONPATH_CONTEXT.parse(child.toString()).read<Any>(column.value)
+                    } else {
+                        val pointer = if (column.value.startsWith("/")) column.value else "/${column.value}"
+                        val valueNode = child.at(pointer)
+
+                        if (valueNode.isValueNode && !valueNode.isNull) {
+                            objectMapper.treeToValue<Any?>(valueNode)
+                        } else {
+                            null
+                        }
+                    }
+
+                    column.key to value
+                }
+            }
+
+        return PageImpl(result, pageable, collection.size().toLong())
+    }
+
+    class InvalidCollectionException : RuntimeException("Invalid collection value!")
```

Comment:
You'll reuse this with the collection case widget as well I think. Could you move this to a separate file?